ethereum/contracts/ETHApp.sol

```diff
@@ -13,8 +14,9 @@ enum ChannelId {
 }
 
 contract ETHApp is RewardSource, AccessControl {
-    using ScaleCodec for uint256;
+    using ScaleCodec for uint128;
     using ScaleCodec for uint32;
+    using SafeCast for uint256;
 
     uint256 public balance;
```

Comment:
There may be a condition in that the balance can exceed the amount that may be stored in uint128.
 As a result, I believe that keeping uint256 is preferable to as per changing to uint128.

Reply:
I'm assuming you're referring to the update of `balance` in `lock()`, which could overflow? That condition should result in the transaction being reverted. Which is fine and would be expected behavior. 

Since we are now on Solidity 0.8.0, safe arithmetic is enabled by default. But we should add unit tests for extra cover.

Reply:
Yup, I am referring to that condition only, with solidity version 0.8.0 and higher will revert the transaction. 
will add a test-case for this and also change to uint128.